filter.go

```diff
@@ -160,6 +163,42 @@ func (csr *CappedSizeRecorder) WriteHeader(statusCode int) {
 	csr.recorder.WriteHeader(statusCode)
 }
 
+func (m Middleware) replacer(input []byte, repl *caddy.Replacer) []byte {
+	pattern := m.compiledSearchRegex
+	if pattern == nil {
+		return input
+	}
+
+	rawReplacement := []byte(m.Replacement) // convert to byte[] directly after reading?
+	if len(rawReplacement) <= 0 {
+		return []byte{}
+	}
+	groups := pattern.FindSubmatch(input)
+	replacement := paramReplacementPattern.ReplaceAllFunc(rawReplacement, func(input2 []byte) []byte {
+		return m.paramReplacer(input2, groups, repl)
+	})
+	return replacement
+}
+
+func (m Middleware) paramReplacer(input []byte, groups [][]byte, repl *caddy.Replacer) []byte {
```

Comment:
Add some document here saying it supports both `{caddy_replacer}` and capture group replacement like `{1}`

Reply:
Done, also added a comment to the README